### PR TITLE
add-devisePullRequest

### PR DESCRIPTION
**Pull request by IvonneBenitesRodriguez**
**About this project**<br/>
This part of the project about building a web app that requires the user to log in.

**For it** :<br/>
I created a featured branch named add-devise
I used devise gem for authentication
Linters were passed.
Gitflow was respected.

### CODE REVIEW

File: db/migrate/20231222204530_add_unique_index_to_email.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class AddUniqueIndexToEmail < ActiveRecord::Migration[7.1]
+  def change
+  end
+end
```
Comment (zieeco) [inline db/migrate/20231222204530_add_unique_index_to_email.rb]: - You've added an empty migration file. Please, reevaluate to remove it.

File: db/migrate/20231222190024_add_email_to_users.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddEmailToUsers < ActiveRecord::Migration[7.1]
+  def change
+   #add_column :users, :email, :string
+  end
+end
```
Comment (zieeco) [inline db/migrate/20231222190024_add_email_to_users.rb]: - The same issue is applicable here as you've added an empty migration file. Please, reevaluate to remove it.

File: db/migrate/20231222154012_add_devise_to_users.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      # t.datetime :current_sign_in_at
+      # t.datetime :last_sign_in_at
+      # t.string   :current_sign_in_ip
+      # t.string   :last_sign_in_ip
+
+      ## Confirmable
+      # t.string   :confirmation_token
+      # t.datetime :confirmed_at
+      # t.datetime :confirmation_sent_at
+      # t.string   :unconfirmed_email # Only if using reconfirmable
```
Comment (zieeco) [inline db/migrate/20231222154012_add_devise_to_users.rb]: - Please, kindly uncomment these codes to make the confirmable work as expected.

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @Team,





Great job so far! 👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- Great job with adding the devise.✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._




## Optional suggestions

- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20231222204530_add_unique_index_to_email.rb (hunk 1 above)
Comment (IvonneBenitesRodriguez) [reply db/migrate/20231222204530_add_unique_index_to_email.rb]: Dear @zieeco 
Very nice to meet you, 
thanks for the suggestion, I appreciate it, 
the file is already removed.
Best wishes,

File: db/migrate/20231222190024_add_email_to_users.rb (hunk 2 above)
Comment (IvonneBenitesRodriguez) [reply db/migrate/20231222190024_add_email_to_users.rb]: Dear @zieeco , 
Thanks for the suggestion, 
I already modified the change you suggested, 
Best wishes,

File: db/migrate/20231222154012_add_devise_to_users.rb (hunk 3 above)
Comment (IvonneBenitesRodriguez) [reply db/migrate/20231222154012_add_devise_to_users.rb]: Dear @zieeco, 
Thanks for the suggestion, 
I already modified it ,
Best wishes,

**Review by Olamarx — APPROVED** [review]
Hi Team,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Changes fixed ✔️
- Confirmable in place ✔️
- Git flow ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**